bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -256,6 +281,24 @@ fn create_yaml(
         IPFamily::IPv4
     };
 
+    let mut eks_zones: Vec<String> = zones.clone().unwrap();
```

Comment:
I would recommend to not use `.unwrap()` directly here as it will panic with a generic error message if `zones` is None. Instead, use .context() or similar error handling to provide a more meaningful error message
```suggestion
    let mut eks_zones: Vec<String> = zones.clone().context("Failed to get availability zones for EKS cluster")?;
```